pkg/metrics.go

```diff
@@ -0,0 +1,56 @@
+package pkg
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	ctrlmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	externalURLs = []string{"https://httpstat.us/503", "https://httpstat.us/200"}
+
+	urlResponseCode = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "reference_addon_sample_response_code",
+			Help: "external url response.",
+		},
+		[]string{"url"},
+	)
+	urlResponseTime = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "reference_addon_sample_response_time",
+			Help: "external url response time.",
+		},
+		[]string{"url"},
+	)
+)
+
+// RegisterMetrics must register metrics in given registry collector
+func RegisterMetrics() {
+	ctrlmetrics.Registry.MustRegister(urlResponseCode)
+	ctrlmetrics.Registry.MustRegister(urlResponseTime)
+}
+
+func AddURLResponseMetrics() {
```

Comment:
```suggestion
func RequestSampleResponseData() {
```

Any name that indicates a request is being made here seems like a better fit.